plutus-ledger/src/Ledger/Tx.hs

```diff
@@ -217,83 +207,84 @@ instance Pretty CardanoTx where
-    (\(SomeTx (C.Tx (C.TxBody C.TxBodyContent {..}) _) _) ->
+    (\(CardanoAPI.SomeTx (C.Tx (C.TxBody C.TxBodyContent {..}) _) _) ->
         txInsReferenceToPlutusTxIns txInsReference)
  where
      txInsReferenceToPlutusTxIns C.TxInsReferenceNone = Set.empty
      txInsReferenceToPlutusTxIns (C.TxInsReference _ txIns) =
-         Set.fromList $ fmap ((`TxIn` Nothing) . CardanoAPI.fromCardanoTxIn) txIns
+         Set.fromList $ fmap ((`CardanoAPI.TxIn` Nothing) . CardanoAPI.fromCardanoTxIn) txIns
 
-getCardanoTxOutRefs :: CardanoTx -> [(V1.Tx.TxOut, V1.Tx.TxOutRef)]
+getCardanoTxOutRefs :: CardanoTx -> [(TxOut, TxOutRef)]
 getCardanoTxOutRefs = onCardanoTx txOutRefs CardanoAPI.txOutRefs
 
-getCardanoTxOutputs :: CardanoTx -> [V1.Tx.TxOut]
+getCardanoTxOutputs :: CardanoTx -> [TxOut]
 getCardanoTxOutputs = fmap fst . getCardanoTxOutRefs
 
-getCardanoTxUnspentOutputsTx :: CardanoTx -> Map V1.Tx.TxOutRef V1.Tx.TxOut
+getCardanoTxUnspentOutputsTx :: CardanoTx -> Map.Map TxOutRef TxOut
 getCardanoTxUnspentOutputsTx = onCardanoTx unspentOutputsTx CardanoAPI.unspentOutputsTx
 
-getCardanoTxSpentOutputs :: CardanoTx -> Set V1.Tx.TxOutRef
-getCardanoTxSpentOutputs = Set.map V1.Tx.txInRef . getCardanoTxInputs
+getCardanoTxSpentOutputs :: CardanoTx -> Set TxOutRef
+getCardanoTxSpentOutputs = Set.map CardanoAPI.txInRef . getCardanoTxInputs
 
-getCardanoTxFee :: CardanoTx -> V1.Value
-getCardanoTxFee = onCardanoTx txFee (\(SomeTx (C.Tx (C.TxBody C.TxBodyContent {..}) _) _) -> CardanoAPI.fromCardanoFee txFee)
+getCardanoTxFee :: CardanoTx -> Value
+getCardanoTxFee = onCardanoTx txFee (\(CardanoAPI.SomeTx (C.Tx (C.TxBody C.TxBodyContent {..}) _) _) -> CardanoAPI.fromCardanoFee txFee)
 
-getCardanoTxMint :: CardanoTx -> V1.Value
-getCardanoTxMint = onCardanoTx txMint (\(SomeTx (C.Tx (C.TxBody C.TxBodyContent {..}) _) _) -> CardanoAPI.fromCardanoMintValue txMintValue)
+getCardanoTxMint :: CardanoTx -> Value
+getCardanoTxMint = onCardanoTx txMint (\(CardanoAPI.SomeTx (C.Tx (C.TxBody C.TxBodyContent {..}) _) _) -> CardanoAPI.fromCardanoMintValue txMintValue)
 
 getCardanoTxValidityRange :: CardanoTx -> SlotRange
 getCardanoTxValidityRange = onCardanoTx txValidRange
-    (\(SomeTx (C.Tx (C.TxBody C.TxBodyContent {..}) _) _) -> CardanoAPI.fromCardanoValidityRange txValidityRange)
+    (\(CardanoAPI.SomeTx (C.Tx (C.TxBody C.TxBodyContent {..}) _) _) -> CardanoAPI.fromCardanoValidityRange txValidityRange)
 
-getCardanoTxData :: CardanoTx -> Map V1.DatumHash V1.Datum
+getCardanoTxData :: CardanoTx -> Map.Map DatumHash Datum
 getCardanoTxData = onCardanoTx txData
-    (\(SomeTx (C.Tx (C.TxBody C.TxBodyContent {..}) _) _) ->
+    (\(CardanoAPI.SomeTx (C.Tx (C.TxBody C.TxBodyContent {..}) _) _) ->
         Map.fromList $ mapMaybe (\(C.TxOut _ _ d _) -> fromCardanoTxOutDatum d) txOuts)
     where
-        fromCardanoTxOutDatum :: C.TxOutDatum C.CtxTx era -> Maybe (V1.DatumHash, V1.Datum)
+      -- FIXME duplicated code
```

Comment:
We already have a `fromCardanoTxOutDatum` somewhere with a slightly different signature.